File: business/voiceroom/src/main/java/cn/rongcloud/voice/model/UiRoomModel.java
package cn.rongcloud.voice.model;

import cn.rongcloud.config.bean.VoiceRoomBean;
import cn.rongcloud.voiceroom.model.RCVoiceRoomInfo;
import io.reactivex.rxjava3.subjects.BehaviorSubject;

public class UiRoomModel {
    private RCVoiceRoomInfo rcRoomInfo;
    private VoiceRoomBean roomBean;
    // 静音标识
    private boolean isMute;

    private BehaviorSubject<UiRoomModel> subject;

    public UiRoomModel(BehaviorSubject<UiRoomModel> subject) {
        this.subject = subject;
    }

    public RCVoiceRoomInfo getRcRoomInfo() {
        return rcRoomInfo;
    }

    public void setRcRoomInfo(RCVoiceRoomInfo rcRoomInfo) {
        this.rcRoomInfo = rcRoomInfo;
        if (null != subject) subject.onNext(this);
    }

    public VoiceRoomBean getRoomBean() {
        return roomBean;
    }

    public void setRoomBean(VoiceRoomBean roomBean) {
        this.roomBean = roomBean;
        if (null != subject) subject.onNext(this);
    }

    public boolean isMute() {
        return isMute;
    }

    public void setMute(boolean mute) {
        isMute = mute;
        if (null != subject) subject.onNext(this);
    }


    public void setSeatCount(int count) {
        if (null != rcRoomInfo)rcRoomInfo.setSeatCount(count);
        if (null != subject) subject.onNext(this);
    }
    public int getSeatCount() {
        return null != rcRoomInfo ? rcRoomInfo.getSeatCount() : 9;
    }

    public void setFreeEnterSeat(boolean freeEnterSeat) {
       if (null != rcRoomInfo)rcRoomInfo.setFreeEnterSeat(freeEnterSeat);
        if (null != subject) subject.onNext(this);
    }

    public boolean isFreeEnterSeat() {
        return null != rcRoomInfo ? rcRoomInfo.isFreeEnterSeat() : false;
    }

    public boolean isLockAll() {
        return null != rcRoomInfo ? rcRoomInfo.isLockAll() : false;
    }


    public boolean isMuteAll() {
        return null != rcRoomInfo ? rcRoomInfo.isMuteAll() : false;
    }
}

File: business/voiceroom/src/main/java/cn/rongcloud/voice/ui/uimodel/UiRoomModel.kt
/*
 * Copyright © 2021 RongCloud. All rights reserved.
 */

package cn.rongcloud.voice.ui.uimodel

import cn.rongcloud.config.bean.VoiceRoomBean
import cn.rongcloud.voiceroom.model.RCVoiceRoomInfo
import io.reactivex.rxjava3.subjects.BehaviorSubject

/**
 * @author gusd
 * @Date 2021/06/23
 */
class UiRoomModel(private val roomInfoSubject: BehaviorSubject<UiRoomModel>) {
    var rcRoomInfo: RCVoiceRoomInfo? = null
        set(value) {
            field = value
            roomInfoSubject.onNext(this)
        }
    var roomBean: VoiceRoomBean? = null
        set(value) {
            field = value
            roomInfoSubject.onNext(this)
        }

    var isMute: Boolean = false
        set(value) {
            field = value
            roomInfoSubject.onNext(this)
        }

    var mSeatCount: Int
        get() {
            return rcRoomInfo?.seatCount ?: 0
        }
        set(value) {
            rcRoomInfo?.let {
                it.seatCount = value
                roomInfoSubject.onNext(this)
            }
        }


    var isFreeEnterSeat: Boolean
        get() {
            return rcRoomInfo?.isFreeEnterSeat ?: false
        }
        set(value) {
            rcRoomInfo?.let {
                it.isFreeEnterSeat = value
                roomInfoSubject.onNext(this)
            }
        }

    var isLockAll: Boolean
        get() {
            return rcRoomInfo?.isLockAll ?: false
        }
        set(value) {
            rcRoomInfo?.let {
                it.isLockAll = value
                roomInfoSubject.onNext(this)
            }
        }

    var isMuteAll: Boolean
        get() {
            return rcRoomInfo?.isMuteAll ?: false
        }
        set(value) {
            rcRoomInfo?.let {
                it.isMuteAll = value
                roomInfoSubject.onNext(this)
            }
        }

    override fun toString(): String {
        return "UiRoomModel(roomInfoSubject=$roomInfoSubject, rcRoomInfo=$rcRoomInfo, roomBean=$roomBean, isMute=$isMute, mSeatCount=$mSeatCount, isFreeEnterSeat=$isFreeEnterSeat, isLockAll=$isLockAll, isMuteAll=$isMuteAll)"
    }


}